Assets/Code/src/Runtime/Core/GameController.cs

```diff
@@ -19,10 +19,12 @@ public class MatchController : IMatchController {
     inputContext = new MatchInputContext(new MatchInput(config));
   }
 
-  public virtual void Update() {
-    var input = InputSource.SampleInput();
-    Assert.IsTrue(input.IsValid);
-    inputContext.Update(input);
+public virtual void Update() {
+    if (!MatchManager.Instance.isControllerLocked){
```

Comment:
MatchController.Update is called mainly from MatchManager, where isControllerLocked is defined. I would advise just wrapping calls to it in MatchManager, not here.